### ISSUE TITLE
PointSetBC accepts a 1-D `values` array and silently broadcasts the error to (N, N)

### ISSUE THREAD

**Issue by msomeya1** [lululxvi/deepxde#2100]
I hit this problem in an inverse problem where I fit some 1D-data at some observation points with `PointSetBC`. Training ran without any error or warning, but the data loss does not drop. It took me a while to realize why; I had read the observed data with `np.loadtxt`, which returns shape `(N,)`, and passed that array straight to `PointSetBC`.

Minimal version:

```python
import os
os.environ["DDE_BACKEND"] = "pytorch"
import numpy as np
import deepxde as dde
import torch

dde.config.set_random_seed(0)

N = 20
points = np.linspace(0, 1, N).reshape(-1, 1)
values = np.sin(2 * np.pi * points).ravel()   # (N,), not (N, 1)

bc = dde.icbc.PointSetBC(points, values, component=0)
err = bc.error(None, None, torch.zeros(N, 1), 0, N)
print("error shape:", tuple(err.shape))

data = dde.data.PDE(dde.geometry.Interval(0, 1), lambda x, y: 0 * y,
                    [bc], num_domain=2)
model = dde.Model(data, dde.nn.FNN([1] + [32] * 3 + [1], "tanh", "Glorot uniform"))
model.compile("adam", lr=1e-3)
model.train(iterations=5000, display_every=1000)
print("Var(values) =", values.var())
```

Output:

```
error shape: (20, 20)
Step      Train loss              Test loss
0         [0.00e+00, 5.03e-01]    [0.00e+00, 5.03e-01]
(omitted)
5000      [0.00e+00, 4.75e-01]    [0.00e+00, 4.75e-01]
Var(values) = 0.475
```

If I use `values.reshape(-1, 1)` instead, the same script converges to ~1e-7 as expected.

I think the problem comes from `PointSetBC.error`.

```python
return outputs[beg:end, self.component : self.component + 1] - self.values
```

`outputs[beg:end, c:c+1]` is `(N, 1)` and `self.values` is `(N,)`, so the subtraction broadcasts to `(N, N)`. The loss then becomes the mean over all pairs,

```
mean_ij (u_i - v_j)^2 = Var(v) + mean_i (u_i - mean(v))^2
```

which is minimized when the network output is spatially constant, with floor `Var(values)`. That matches the 0.475 plateau above, and in my real problem the plateau matched `0.5(Var(u_obs) + Var(v_obs))` to four digits, which is how I eventually tracked it down.

The docstring says `values` should be "A scalar or a 2D-array of values", and `DirichletBC` and `PointSetOperatorBC` both do check the shape of what they get, but `PointSetBC` do not.

I think it is helpful if `PointSetBC` does that kind of check too.
(However, I did not read the entire DeepXDE codebase. I would appreciate it if anyone could suggest other good solutions.)

Versions: DeepXDE 1.12.1 (the relevant code looks the same on master), PyTorch 2.4.1+cu121,
Python 3.10.12, NumPy 1.26.4.

**Comment by AyushG-1210** [comment]
Opened a fix in PR #2101  

The issue happens because `PointSetBC` wasn't validating the shape of `values` before passing it to `error()`, causing implicit broadcasting to `(N, N)` when subtracting against `(N, 1)` network outputs. The PR adds strict shape validation in `__init__` (similar to `DirichletBC` and `PointSetOperatorBC`), which now throws a descriptive `RuntimeError` advising the use of `.reshape(-1, 1)` if a 1D array is passed.
